src/Frontend/Modules/MailMotor/Actions/Subscribe.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Frontend\Modules\MailMotor\Actions;
+
+/*
+ * This file is part of the Fork CMS MailMotor Module from SIESQO.
+ *
+ * For the full copyright and license information, please view the license
+ * file that was distributed with this source code.
+ */
+
+use Frontend\Core\Engine\Base\Block as FrontendBaseBlock;
+use Frontend\Core\Engine\Language;
+use Frontend\Core\Engine\Form as FrontendForm;
+use Frontend\Core\Engine\Navigation as FrontendNavigation;
+use Frontend\Modules\MailMotor\Engine\Model as FrontendMailMotorModel;
+use MailMotor\Bundle\MailMotorBundle\Exception\NotImplementedException;
+
+/**
+ * This is the Subscribe-action for our MailMotor
+ *
+ * @author Jeroen Desloovere <jeroen@siesqo.be>
+ */
+class Subscribe extends FrontendBaseBlock
+{
+    /**
+     * FrontendForm instance
+     *
+     * @var	FrontendForm
+     */
+    private $frm;
+
+    /**
+     * Execute the extra
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        parent::execute();
+        $this->loadTemplate();
+        $this->loadForm();
+        $this->validateForm();
+        $this->parse();
+    }
+
+    /**
+     * Load the form
+     *
+     * @return void
+     */
+    private function loadForm()
+    {
+        // create the form
+        $this->frm = new FrontendForm('mailMotorSubscribeForm');
+
+        // define email
+        $email = null;
+
+        // request contains an email
+        if ($this->get('request')->request->get('email') != null) {
+            $email = $this->get('request')->request->get('email');
+        }
```

Comment:
this looks like `$this->get('request')->request->get('email')` returns null if it isn't available in the request? This means that th define and if aren't necessary.